lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -491,31 +520,47 @@ export class UserAgentApplication {
         scopes = this.filterScopes(scopes);
       }
 
-      // Extract ADAL id_token if it exists
-      let idTokenObject;
-      idTokenObject = this.extractADALIdToken();
+      // construct extraQueryParams string from the request
+      let extraQueryParameters = Utils.constructExtraQueryParametersString(request.extraQueryParameters);
+      extraQueryParameters = Utils.addPromptParameter(extraQueryParameters, request.prompt);
+      let user = this.getUser();
 
-      // silent login if ADAL id_token is retrieved successfully - SSO
-      if (idTokenObject && !scopes) {
-        this.logger.info("ADAL's idToken exists. Extracting login information from ADAL's idToken ");
-        extraQueryParameters = Utils.constructUnifiedCacheExtraQueryParameter(idTokenObject, extraQueryParameters);
+      // if the developer provides one of these, give preference to developer choice
+      if (request.account || request.sid || request.loginHint) {
+        extraQueryParameters = Utils.constructUnifiedCacheExtraQueryParameter(extraQueryParameters, request, null);
 
-        this.silentLogin = true;
-        this.acquireTokenSilent([this.clientId], this.authority, this.getUser(), extraQueryParameters)
-        .then((idToken) => {
-          this.silentLogin = false;
-          this.logger.info("Unified cache call is successful");
-          // TODO: Change resolve to return AuthResponse object
-          resolve(idToken);
-        }, (error) => {
-          this.silentLogin = false;
-          this.logger.error("Error occurred during unified cache ATS");
-          this.loginPopupHelper(resolve, reject, scopes, extraQueryParameters);
-        });
+         // if user is not provided, we pass null
+         this.loginPopupHelper(user, resolve, reject, scopes, extraQueryParameters);
       }
-      // else proceed with login
+      // else handle the library data
       else {
-        this.loginPopupHelper(resolve, reject, scopes, extraQueryParameters );
+        // Extract ADAL id_token if it exists
+        let idTokenObject = this.extractADALIdToken();
```

Comment:
if not in this pr, in a follow up lets constify what we can

Reply:
can we name clearer so its clear it is an Adal Token

Reply:
Renamed.